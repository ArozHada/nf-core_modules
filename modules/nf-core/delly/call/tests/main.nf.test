nextflow_process {

    name "Test Process DELLY_CALL"
    script "../main.nf"
    process "DELLY_CALL"

    tag "modules"
    tag "modules_nfcore"
    tag "delly"
    tag "delly/call"

    test("homo_sapiens - [bam,bai]") {

        when {
            process {
                """

                input[0] = [
                        [ id:'test' ],
                        file(params.test_data['homo_sapiens']['illumina']['test_paired_end_recalibrated_sorted_bam'], checkIfExists: true),
                        file(params.test_data['homo_sapiens']['illumina']['test_paired_end_recalibrated_sorted_bam_bai'], checkIfExists: true),
                        [],
                        [],
                        [],
                        ]
                input[1] = [ [ id:'test' ], file(params.test_data['homo_sapiens']['genome']['genome_21_fasta'], checkIfExists: true)]
                input[2] = [ [ id:'test' ], file(params.test_data['homo_sapiens']['genome']['genome_21_fasta_fai'], checkIfExists: true)]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("homo_sapiens - [cram,crai]") {

        config "./nextflow_call_cram.config"

        when {
            process {
                """

                input[0] = [
                        [ id:'test' ],
                        file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_cram'], checkIfExists: true),
                        file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_cram_crai'], checkIfExists: true),
                        [],
                        [],
                        [],
                        ]
                input[1] = [ [ id:'test' ], file(params.test_data['homo_sapiens']['genome']['genome_fasta'], checkIfExists: true)]
                input[2] = [ [ id:'test' ], file(params.test_data['homo_sapiens']['genome']['genome_fasta_fai'], checkIfExists: true)]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("homo_sapiens - [cram,crai] - exclude_regions - bed") {

        when {
            process {
                """

                input[0] = [
                        [ id:'test' ],
                        file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_cram'], checkIfExists: true),
                        file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_cram_crai'], checkIfExists: true),
                        [],
                        [],
                        file(params.test_data['homo_sapiens']['genome']['genome_21_multi_interval_bed'], checkIfExists: true),
                        ]
                input[1] = [ [ id:'test' ], file(params.test_data['homo_sapiens']['genome']['genome_fasta'], checkIfExists: true)]
                input[2] = [ [ id:'test' ], file(params.test_data['homo_sapiens']['genome']['genome_fasta_fai'], checkIfExists: true)]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("homo_sapiens - [cram,crai] - genotype [vcf.gz, vcf.gz.tbi]") {

        config "./nextflow_call_genotype.config"

        when {
            process {
                """

                input[0] = [
                        [ id:'test' ],
                        file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_cram'], checkIfExists: true),
                        file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_cram_crai'], checkIfExists: true),
                        file(params.test_data['homo_sapiens']['illumina']['test_haplotc_cnn_vcf_gz'], checkIfExists: true),
                        file(params.test_data['homo_sapiens']['illumina']['test_haplotc_cnn_vcf_gz_tbi'], checkIfExists: true),
                        []
                        ]
                input[1] = [ [ id:'test' ], file(params.test_data['homo_sapiens']['genome']['genome_fasta'], checkIfExists: true)]
                input[2] = [ [ id:'test' ], file(params.test_data['homo_sapiens']['genome']['genome_fasta_fai'], checkIfExists: true)]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

     test("homo_sapiens-stub") {

        options "-stub-run"

        when {
            process {
                """
                input[0] = [
                        [ id:'test' ],
                        file(params.test_data['homo_sapiens']['illumina']['test_paired_end_recalibrated_sorted_bam'], checkIfExists: true),
                        file(params.test_data['homo_sapiens']['illumina']['test_paired_end_recalibrated_sorted_bam_bai'], checkIfExists: true),
                        [],
                        [],
                        [],
                        ]
                input[1] = [ [ id:'test' ], file(params.test_data['homo_sapiens']['genome']['genome_fasta'], checkIfExists: true)]
                input[2] = [ [ id:'test' ], file(params.test_data['homo_sapiens']['genome']['genome_fasta_fai'], checkIfExists: true)]

                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(
                    file(process.out.bcf[0][1]).name,
                    file(process.out.csi[0][1]).name).match() }
            )
        }

    }

}
