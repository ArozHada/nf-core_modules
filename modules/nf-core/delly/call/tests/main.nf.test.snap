{
    "homo_sapiens - [cram,crai]": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.vcf.gz:md5,b27408b3a42049d8c180761265ce44bf"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test.vcf.gz.tbi:md5,e7180bb953d2bd657c420a5f76a7164d"
                    ]
                ],
                "2": [
                    "versions.yml:md5,913e0d1c14eef3de3871438a25713a48"
                ],
                "bcf": [
                    [
                        {
                            "id": "test"
                        },
                        "test.vcf.gz:md5,b27408b3a42049d8c180761265ce44bf"
                    ]
                ],
                "csi": [
                    [
                        {
                            "id": "test"
                        },
                        "test.vcf.gz.tbi:md5,e7180bb953d2bd657c420a5f76a7164d"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,913e0d1c14eef3de3871438a25713a48"
                ]
            }
        ],
        "timestamp": "2023-11-29T17:42:17.942812572"
    },
    "homo_sapiens - [cram,crai] - genotype [vcf.gz, vcf.gz.tbi]": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.vcf.gz:md5,b27408b3a42049d8c180761265ce44bf"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test.vcf.gz.tbi:md5,e7180bb953d2bd657c420a5f76a7164d"
                    ]
                ],
                "2": [
                    "versions.yml:md5,913e0d1c14eef3de3871438a25713a48"
                ],
                "bcf": [
                    [
                        {
                            "id": "test"
                        },
                        "test.vcf.gz:md5,b27408b3a42049d8c180761265ce44bf"
                    ]
                ],
                "csi": [
                    [
                        {
                            "id": "test"
                        },
                        "test.vcf.gz.tbi:md5,e7180bb953d2bd657c420a5f76a7164d"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,913e0d1c14eef3de3871438a25713a48"
                ]
            }
        ],
        "timestamp": "2023-11-29T17:42:46.525120871"
    },
    "homo_sapiens - [bam,bai]": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bcf:md5,2ff1b1b6a2f3a81ca2fcc6a8836f6389"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bcf.csi:md5,66545c77fc791a02dadc7fc252d04635"
                    ]
                ],
                "2": [
                    "versions.yml:md5,913e0d1c14eef3de3871438a25713a48"
                ],
                "bcf": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bcf:md5,2ff1b1b6a2f3a81ca2fcc6a8836f6389"
                    ]
                ],
                "csi": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bcf.csi:md5,66545c77fc791a02dadc7fc252d04635"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,913e0d1c14eef3de3871438a25713a48"
                ]
            }
        ],
        "timestamp": "2023-11-29T17:42:00.048727722"
    },
    "homo_sapiens - [cram,crai] - exclude_regions - bed": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bcf:md5,9e83cc9f6255e3004c66e403ed65bb82"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bcf.csi:md5,a2ba4f0b32a6ea857ec8a5b3068f168f"
                    ]
                ],
                "2": [
                    "versions.yml:md5,913e0d1c14eef3de3871438a25713a48"
                ],
                "bcf": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bcf:md5,9e83cc9f6255e3004c66e403ed65bb82"
                    ]
                ],
                "csi": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bcf.csi:md5,a2ba4f0b32a6ea857ec8a5b3068f168f"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,913e0d1c14eef3de3871438a25713a48"
                ]
            }
        ],
        "timestamp": "2023-11-29T17:42:29.052293068"
    },
    "homo_sapiens-stub": {
        "content": [
            "test.bcf",
            "test.bcf.csi"
        ],
        "timestamp": "2023-11-29T17:43:00.124658313"
    }
}